vault/seal.go

```diff
@@ -197,8 +225,10 @@ func (d *defaultSeal) BarrierConfig(ctx context.Context) (*SealConfig, error) {
 		conf.Type = d.BarrierType().String()
 	case d.BarrierType().String():
 	default:
-		d.core.logger.Error("barrier seal type does not match expected type", "barrier_seal_type", conf.Type, "loaded_seal_type", d.BarrierType())
-		return nil, fmt.Errorf("barrier seal type of %q does not match expected type of %q", conf.Type, d.BarrierType())
+		if !(conf.Type != wrapping.WrapperTypeShamir.String() && d.unsealKeyPath == recoveryUnsealKeyPath) {
```

Comment:
You could eliminate the `if` by replacing `default` with the condition.  Also, would it be clearer if you expanded the negation and eliminated the parens, i.e. `conf.Type == wrapping.WrapperTypeShamir.String(), d.unsealKeyPath != recoveryUnsealKeyPath`?

Reply:
The switch is over conf.Type, so not quite that simple.  I'll invert the predicate though, simple enough.